namespace/builder.go

```diff
@@ -0,0 +1,126 @@
+package namespace
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// OpenFunc opens files
+type OpenFunc func(path string) (io.Reader, error)
+
+// Builder helps building plan9 namespaces
+type Builder struct {
+	dir  string
+	file File
+
+	open OpenFunc
+}
+
+func open1(path string) (io.Reader, error) { return os.Open(path) }
+
+// NewBuilder returns the default runner
+func NewBuilder() (*Builder, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	return &Builder{
+		dir:  wd,
+		open: open1,
+	}, nil
+}
+
+func newrunner(wd string, b OpenFunc) (*Builder, error) {
+	return &Builder{
+		dir:  wd,
+		open: b,
+	}, nil
+}
+
+// Parse takes a path and parses the namespace file
+func (b *Builder) Parse(file string) error {
+	var f io.Reader
+	var err error
+	f, err = b.open(file)
+	if err != nil {
+		return err
+	}
+	b.file, err = Parse(f)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Run takes a namespace and runs commands defined in the namespace file
+func (b *Builder) buildns(ns Namespace) error {
+	for _, c := range b.file {
+		var err error
+		args := []string{}
+		for _, arg := range c.args {
+			args = append(args, os.ExpandEnv(arg))
+		}
+
+		switch syzcall(c.syscall) {
+		case BIND:
+			err = ns.Bind(args[0], args[1], int(c.flag))
```

Comment:
Has the number of args been validated by this time?

Reply:
yes.